src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -0,0 +1,111 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import java.io.Writer;
+import java.io.StringWriter;
+
+public class PipelineConsoleViewAction extends AbstractPipelineViewAction {
+    private static final Logger LOGGER = Logger.getLogger(PipelineConsoleViewAction.class.getName());
+    private final WorkflowRun target;
+    public PipelineConsoleViewAction(WorkflowRun target) {
+        super(target);
+        this.target = target;
+    }
+
+    @Override
+    public String getDisplayName() {
+        LOGGER.log(Level.INFO, "PipelineConsoleViewAction getDisplayName called.");
+        return "Pipeline Console";
+    }
+
+    @Override
+    public String getUrlName() {
+        LOGGER.log(Level.INFO, "PipelineConsoleViewAction getUrlName called.");
+        return "pipeline-console";
+    }
+
+    // Consider making this return all steps (don't accept the node)
+    // If the steps include the parent ID (or be indexed by it), then I can save a list of steps to the 
+    // React component state. Then I can cross-reference this when adding the child nodes instead of
+    // making a fetch call - which seems impossible to wait for the result of (maybe fetch is the wrong thing to use?)
+    @WebMethod(name = "steps")
+    public void getSteps(StaplerRequest req, StaplerResponse rsp) throws IOException {
+        LOGGER.log(Level.INFO, "PipelineConsoleViewAction getSteps called.");
+        String nodeId = req.getParameter("nodeId");
+        if (nodeId != null) {
+            LOGGER.log(Level.INFO, "PipelineConsoleViewAction getSteps passed nodeId '" + nodeId + "'.");
+            PipelineStepApi stepApi = new PipelineStepApi(target, nodeId);
+            ObjectMapper mapper = new ObjectMapper();
+            LOGGER.log(Level.INFO, "Steps: '" + mapper.writeValueAsString(stepApi.getSteps()) + "'.");
+            rsp.getWriter().append(mapper.writeValueAsString(stepApi.getSteps()));
+        } else {
+            LOGGER.log(Level.INFO, "PipelineConsoleViewAction getSteps not passed nodeId.");
+            rsp.getWriter().append("{\"text\": \"Error getting console text\"}");
+        }
+    }
+
+    @WebMethod(name = "consoleOutput")
+    public void getConsoleOutput(StaplerRequest req, StaplerResponse rsp) throws IOException {
+        LOGGER.log(Level.INFO, "PipelineConsoleViewAction getConsoleOutput called.");
+        String nodeId = req.getParameter("nodeId");
+        if (nodeId != null) {
+            LOGGER.log(Level.INFO, "PipelineConsoleViewAction getConsoleOutput passed nodeId.");
+            String nodeConsoleText = getLogForNode(nodeId);
+            if (nodeConsoleText != null) {
+                rsp.getWriter().append(nodeConsoleText);
+            } else {
+                rsp.getWriter().append("{\"text\": \"No console output for node: " + nodeId + "\"}");
```

Comment:
I recommend just returning plain text here rather than wrapping it in a json object.
my test maven build failed here because what was returned was not proper json